spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java

```diff
@@ -65,9 +69,10 @@ public StreamTDigestReader(DataInputStream in) {
    * Consume all data from the stream create a list of the measurement batches as they were
    * read from the stream.
    */
-  static List<List<TDigestMeasurement>> readAll(InputStream in) throws IOException {
+  static List<List<TDigestMeasurement>> readAll(Registry registry, InputStream in)
```

Comment:
Should this method be made non-static and use the registry passed into the class constructor?  The constructor would need to be changed to keep a direct reference to the Registry and not just use it to construct the Json object.